src/components/Editable.js

```diff
@@ -88,6 +89,58 @@ export const Editable = connect()(({ isEditing, thoughtsRanked, contextChain, sh
     }
   }
 
+  const onChangeHandler = e => {
+
+    const state = store.getState()
+
+    // NOTE: When Subthought components are re-rendered on edit, change is called with identical old and new values (?) causing an infinite loop
+    const newValue = he.decode(strip(e.target.value))
+
+    // safari adds <br> to empty contenteditables after editing, so strip thnem out
+    // make sure empty thoughts are truly empty
+    if (ref.current && newValue.length === 0) {
+      ref.current.innerHTML = newValue
+    }
+
+    if (newValue !== oldValue) {
+      const thought = getThought(oldValue)
+      if (thought) {
+        dispatch({ type: 'existingThoughtChange', context, showContexts, oldValue, newValue, rankInContext: rank, thoughtsRanked, contextChain })
+
+        // rerender so that triple dash is converted into horizontal rule
+        // otherwise nothing would be rerendered because the thought is still being edited
+        if (isDivider(newValue)) {
+          dispatch({ type: 'render' })
+        }
+
+        // store the value so that we have a transcendental head when it is changed
+        oldValue = newValue
+
+        const { tutorialChoice, tutorialStep } = state.settings
+        if (newValue && (
+          (
+            Math.floor(tutorialStep) === TUTORIAL2_STEP_CONTEXT1_PARENT &&
+            newValue.toLowerCase() === TUTORIAL_CONTEXT1_PARENT[tutorialChoice].toLowerCase()
+          ) || (
+            Math.floor(tutorialStep) === TUTORIAL2_STEP_CONTEXT2_PARENT &&
+            newValue.toLowerCase() === TUTORIAL_CONTEXT2_PARENT[tutorialChoice].toLowerCase()
+          ) || (
+            (
+              Math.floor(tutorialStep) === TUTORIAL2_STEP_CONTEXT1 ||
+              Math.floor(tutorialStep) === TUTORIAL2_STEP_CONTEXT2
+            ) &&
+            newValue.toLowerCase() === TUTORIAL_CONTEXT[tutorialChoice].toLowerCase()
+          )
+        )) {
+          tutorialNext()
+        }
+      }
+    }
+  }
+
+  // using useRef hook to store throttled function so that it can persist even between component re-renders, so that throttle.flush method can be used properly
+  const throttledChangeRef = useRef(throttle(onChangeHandler, 800))
```

Comment:
Let's add `const EDIT_THROTTLE = 800` to `constants.js`.

Reply:
sure